### ISSUE TITLE
Autostart before login

### ISSUE THREAD

**Issue by hogAlert** [azalinux/realvnc-server-aarch64-manjaro#1]
Thanks for this very useful project and the accompanying youtube video.

I have enabled the service as instructed, but this seems only to start after user logon?

Is any way to configure the server to start *before* logon to allow login to a headless machine (this is how it is on Raspberry Pi OS).

On further investigation, it seems like the problem might (also?) be that there is no (wireless) internet access at the logon stage?
I say this because I noticed that If I log-out then a VNC server window pops up (visible on the directly connected display) saying 'no connectivity'.

By the way the direct link in the youtube description wget command is bad, it points to:

https://github.com/azalinux/realvnc-server-aarch64-manjaro/releases/download/v6.7.2.43081/realvnc-vnc-server-6.7.2.43081-1-aarch64.pkg.tar.zst

instead of:

https://github.com/azalinux/realvnc-server-aarch64-manjaro/releases/download/realvnc-vnc-server-6.7.2.43081-1-aarch64/realvnc-vnc-server-6.7.2.43081-1-aarch64.pkg.tar.zst

Thanks again for a useful project!

**Comment by azalinux** [comment]
Heya,

Thanks for the info file location issue.  I've updated the link location in youtube to the correct URL :)

Thats a bit weird with what you are describing - since the realvnc server gets started via SystemD this should enable at system boot rather than a physical user login.

I'd be interested to know your current Pi setup and if you start your Pi with a graphical target or just terminal..?  
For reference, my main Pi4 is headless (ie no monitor attached) but it still runs xorg & XFCE by default.  You need this in your Pi config.txt file (normally in /boot ):

hdmi_force_hotplug=1

without this in Arch, you will have issues with headless.

ive only ever used Ethernet with my Pi , so the wireless phenomenon i haven't dealt with. 

I'll try and see if I can replicate what your describing but in the meantime, let me know what manjaro version you are using and whether the system boots into a command line or graphicsl target ..?

cheers
Aaron

**Comment by hogAlert** [comment]
Hi, It's a manjaro/xfce system, I was using a fresh install from the manjaro website (v21.04 as of yesterday ) so yes it's GUI.
I'm using a Pi4 (4GB).
I think the problem might be that WiFi doesn't connect until the user logs in (perhaps the creds are stored per user?), so this might be outside of the scope of this project somewhat (although it renders VNC a bit useless unfortunately since I imagine at lot of uses of VNC are for headless systems and WiFi as a connection method is not uncommon..)

**Comment by azalinux** [comment]
no worries , i'll give it a try with wifi and see if the same thing happens and i'll let u know tomorrow

**Comment by azalinux** [comment]
UPDATE!

Yes you are correct - in Manjaro the WIFI does not connect automatically before user login!

I've managed to get it working - check out the following and give it a try :)

To get around this, Right hand click on the NetworkManager / WIFI icon in the system tray and click Edit Connections.

Select the WIFI name and click the little 'Cog' (settings) icon near the bottom of the window.  This will bring up the Settings for the WIFI - Click the General Tab then tick 'All Users may Connect To this network" .  Then click WIFI Security Tab and where it has the Password - on the extreme right of the password field it has a little icon - click on it once and make sure "Store The password for all users" is selected. 

Press Save down the bottom and reboot the pi.  This will fire up the wifi before user login. 

I've tested this on my Pi on Manjaro MATE - i'll try it on XFCE in the morning.  It should be the same outcome as both use the standard "NetworkManager" module for network access and config.
